Modules/Loadable/Markups/VTKWidgets/vtkSlicerROIRepresentation2D.cxx

```diff
@@ -192,11 +191,13 @@ void vtkSlicerROIRepresentation2D::UpdateCubeSourceFromMRML(vtkMRMLMarkupsROINod
 
   double sideLengths[3] = { 0.0, 0.0, 0.0 };
   roiNode->GetSizeWorld(sideLengths);
-  cubeSource->SetXLength(sideLengths[0]);
-  cubeSource->SetYLength(sideLengths[1]);
-  cubeSource->SetZLength(sideLengths[2]);
-
-  this->ROIOutlineActor->SetVisibility(vtkMath::Norm(sideLengths) >= 1e-6);
+  // Add a small amount to each axis so that the ROI is still visible when the slice view is parallel to
+  // and intersecting with one of the faces.
+  double epsilon = 1e-5;
```

Comment:
1e-5 might not be small enough if the slice spacing is very small (e.g., someone loads a microscopy image and keep the length unit in mm). Could we use some spacing values from the slice node instead of an absolute distance value?

Reply:
Ok, I've made the distance relative to the slice spacing.